content/Snakemake_intro.md

```diff
@@ -12,27 +26,471 @@ Example usecase from GUI -> multiple scripts run manually -> bash script to run
+```bash
+rule print_content:
+    input:
+        "hello.txt"
+    script:
+        "read_and_print.py"
+```
+In the Python file, we have access to the `snakemake` object, which allows us to get access to input and output files! We do not need to parse any command line arguments. The access is granted because of the `script` keyword, which works like a wrapper around the script. Similar integrations are available for other languages as well: R, Markdown, Julia, Rust, Bash and Jupyter Notebook. See the current list in the [docs](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#external-scripts)
+
+
+### Dependencies between rules
+So far, we only examined `Snakefile`s with only one rule. Let's have a look at a bigger workflow.
+```bash
+rule concatenate:
+    input:
+        expand("data/file{n}.txt", n=[1, 2, 3])
+    output:
+        "results/concatenated.txt"
+    shell:
+        "cat {input} > {output}"
+
+rule count_words:
+    input:
+        "results/concatenated.txt"
+    output:
+        "results/word_count.txt"
+    shell:
+        "wc -w {input} > {output}"
+    default_target: True
+```
+This workflow concatenates three input files (`data/file1`, `data/file2`, `data/file3`) and then counts how many words there are. The `Snakefile` consists of three rules. The dependencies between rules are defined by the input and output filenames. The rule `concatenate` generates the output file: `results/concatenated.txt`, which then is used as a input to the `count_words` rule. 
+
+Snakemake uses top-down approach to automatically resolve the dependencies between rules. That means, Snakemake will start from the last rule (so-called target rule) and go backwards, searching for the input needed to execute that step. By default, the target rule is the first rule in the Snakefile. In this case, we want to run the `count_words` rule as the target (first we concatenate, then we count the words). We have to somehow tell Snakemake what is the final output of this workflow. To achieve that, a common practice is adding a mock `rule all` at the beginning of the `Snakefile` that encapsulates the final outputs of the workflow:
+```bash
+rule all:
+    input:
+        "results/word_count.txt"
+
+rule concatenate:
+    input:
+        expand("data/file{n}.txt", n=[1, 2, 3])
+    output:
+        "results/concatenated.txt"
+    shell:
+        "cat {input} > {output}"
+
+rule count_words:
+    input:
+        "results/concatenated.txt"
+    output:
+        "results/word_count.txt"
+    shell:
+        "wc -w {input} > {output}"
+```
+
+Snakemake will see that we want the `results/word_count.txt` file, and will search in the `Snakefile` to see what rule produces this exact output. Then, it will repeat this process recursively until all dependencies are resolved. By doing so, so-called directed acyclic graph (DAG) is created. It is a step-by-step execution plan for Snakemake. We can see and analyze this graph by calling Snakemake with `--dag` flag:
```

Comment:
```suggestion
Snakemake will see that we want the `results/word_count.txt` file, and will search in the `Snakefile` to see what rule produces this exact output. Then, it will repeat this process recursively until all dependencies are resolved. By doing so, a so-called directed acyclic graph (DAG) is created. It is a step-by-step execution plan for Snakemake. We can see and analyze this graph by calling Snakemake with `--dag` flag:
```